### PR TITLE
[incubator-kie-issues-1682] Test cases in kogito repos GHA fail with …

### PR DESCRIPTION
**Pull request by tkobayas**
…"pull access denied for vectorized/redpanda"

### Issue:
- https://github.com/apache/incubator-kie-issues/issues/1682

### Cross repo PRs
- https://github.com/apache/incubator-kie-kogito-runtimes/pull/3812 (This one)
- https://github.com/apache/incubator-kie-kogito-apps/pull/2162
- https://github.com/apache/incubator-kie-kogito-examples/pull/2043

---
- [x] You have read the [contributors guide](CONTRIBUTING.md)
- [x] Your code is properly formatted according to [this configuration](https://github.com/apache/incubator-kie-kogito-runtimes/tree/main/kogito-build/kogito-ide-config)
- [x] Pull Request title is properly formatted: `Issue-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting main: `[0.9.x] Issue-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: kogito-build/kogito-build-no-bom-parent/pom.xml [hunk 1]
```diff
@@ -50,7 +50,7 @@
     <!-- container images for testing -->
     <container.image.infinispan>infinispan/server:${version.org.infinispan}</container.image.infinispan>
     <container.image.keycloak>quay.io/keycloak/keycloak:${version.org.keycloak}</container.image.keycloak>
-    <container.image.kafka>vectorized/redpanda:v21.11.8</container.image.kafka>
+    <container.image.kafka>redpandadata/redpanda:v24.3.1</container.image.kafka>
```
Comment (tkobayas) [inline kogito-build/kogito-build-no-bom-parent/pom.xml:53]: Chose the latest version in https://hub.docker.com/r/redpandadata/redpanda/tags

`v21.11.8` doesn't exist in redpandadata/redpanda.

**Comment by tkobayas** [comment]
Hmm, it's failing... I thought it was successful in my local env. However, I get the same error locally, too.

I'm calling it a day, so it's great if someone can take over the investigation. 
```
2024-12-11T08:11:04.6482206Z 2024-12-11 08:11:04,622 INFO  [tc.red.3.1] (pool-3-thread-1) Creating container for image: redpandadata/redpanda:v24.3.1
2024-12-11T08:11:04.6483689Z 2024-12-11 08:11:04,638 INFO  [tc.red.3.1] (pool-3-thread-1) Container redpandadata/redpanda:v24.3.1 is starting: 361588c8a4c7db92e4dc30bc24b7c92188d09b0da750155ae5f2d44063ba70b8
2024-12-11T08:26:05.0072253Z 2024-12-11 08:26:04,952 ERROR [tc.red.3.1] (pool-3-thread-1) Could not start container: java.lang.IllegalStateException: Wait strategy failed. Container exited with code 1
2024-12-11T08:26:05.0073679Z 	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:533)
2024-12-11T08:26:05.0074627Z 	at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:354)
2024-12-11T08:26:05.0075630Z 	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)
2024-12-11T08:26:05.0076651Z 	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:344)
2024-12-11T08:26:05.0077604Z 	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:330)
2024-12-11T08:26:05.0078876Z 	at org.kie.kogito.testcontainers.KogitoKafkaContainer.start(KogitoKafkaContainer.java:72)
2024-12-11T08:26:05.0080222Z 	at org.kie.kogito.test.resources.ConditionalQuarkusTestResource.start(ConditionalQuarkusTestResource.java:69)
2024-12-11T08:26:05.0081594Z 	at org.kie.kogito.testcontainers.quarkus.KafkaQuarkusTestResource.start(KafkaQuarkusTestResource.java:68)
2024-12-11T08:26:05.0082865Z 	at io.quarkus.test.common.TestResourceManager$TestResourceEntryRunnable.run(TestResourceManager.java:500)
2024-12-11T08:26:05.0084009Z 	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804)
2024-12-11T08:26:05.0085055Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2024-12-11T08:26:05.0086111Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2024-12-11T08:26:05.0086840Z 	at java.base/java.lang.Thread.run(Thread.java:840)
2024-12-11T08:26:05.0087895Z Caused by: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for log output matching '.*Started Kafka API server.*'
2024-12-11T08:26:05.0089406Z 	at org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy.waitUntilReady(LogMessageWaitStrategy.java:47)
2024-12-11T08:26:05.0090926Z 	at org.testcontainers.containers.wait.strategy.AbstractWaitStrategy.waitUntilReady(AbstractWaitStrategy.java:52)
2024-12-11T08:26:05.0092401Z 	at org.testcontainers.containers.GenericContainer.waitUntilContainerStarted(GenericContainer.java:909)
2024-12-11T08:26:05.0093621Z 	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:500)
2024-12-11T08:26:05.0094351Z 	... 12 more
2024-12-11T08:26:05.0094529Z 
2024-12-11T08:26:05.0094912Z 2024-12-11 08:26:04,979 ERROR [tc.red.3.1] (pool-3-thread-1) Log output from the failed container:
2024-12-11T08:26:05.0096394Z WARNING: This is a setup for development purposes only; in this mode your clusters may run unrealistically fast and data can be corrupted any time your computer shuts down uncleanly.
2024-12-11T08:26:05.0098557Z Error: invalid host "PLAINTEXT://361588c8a4c7:29092" does not match "host", nor "host:port", nor "scheme://host:port"
2024-12-11T08:26:05.0099397Z Usage:
2024-12-11T08:26:05.0099682Z   rpk redpanda start [flags]
2024-12-11T08:26:05.0099939Z 
2024-12-11T08:26:05.0100052Z Flags:
2024-12-11T08:26:05.0100514Z       --advertise-kafka-addr strings        A comma-separated list of
2024-12-11T08:26:05.0101156Z                                             Kafka addresses to advertise
2024-12-11T08:26:05.0101683Z                                             (scheme://host:port|name)
2024-12-11T08:26:05.0102503Z       --advertise-pandaproxy-addr strings   A comma-separated list of
2024-12-11T08:26:05.0103120Z                                             Pandaproxy addresses to
2024-12-11T08:26:05.0103610Z                                             advertise (scheme://host:port|name)
2024-12-11T08:26:05.0104284Z       --advertise-rpc-addr string           The advertised RPC address
2024-12-11T08:26:05.0104918Z                                             (host:port)
2024-12-11T08:26:05.0105456Z       --check                               When set to false will disable
2024-12-11T08:26:05.0106021Z                                             system checking before
2024-12-11T08:26:05.0106499Z                                             starting redpanda (default true)
2024-12-11T08:26:05.0106976Z   -h, --help                                Help for start
2024-12-11T08:26:05.0107523Z       --install-dir string                  Directory where redpanda has
2024-12-11T08:26:05.0108105Z                                             been installed
2024-12-11T08:26:05.0108674Z       --kafka-addr strings                  A comma-separated list of
2024-12-11T08:26:05.0109257Z                                             Kafka listener addresses to
2024-12-11T08:26:05.0109719Z                                             bind to (scheme://host:port|name)
2024-12-11T08:26:05.0110394Z       --mode string                         Mode sets well-known
2024-12-11T08:26:05.0111058Z                                             configuration properties for
2024-12-11T08:26:05.0111558Z                                             development or test
2024-12-11T08:26:05.0112038Z                                             environments; use --mode help
2024-12-11T08:26:05.0112550Z                                             for more info
2024-12-11T08:26:05.0113162Z       --pandaproxy-addr strings             A comma-separated list of
2024-12-11T08:26:05.0113799Z                                             Pandaproxy listener addresses
2024-12-11T08:26:05.0114289Z                                             to bind to
2024-12-11T08:26:05.0114795Z                                             (scheme://host:port|name)
2024-12-11T08:26:05.0115403Z       --rpc-addr string                     The RPC address to bind to
2024-12-11T08:26:05.0115969Z                                             (host:port)
2024-12-11T08:26:05.0116617Z       --schema-registry-addr strings        A comma-separated list of
2024-12-11T08:26:05.0117273Z                                             Schema Registry listener
2024-12-11T08:26:05.0117782Z                                             addresses to bind to
2024-12-11T08:26:05.0118303Z                                             (scheme://host:port|name)
2024-12-11T08:26:05.0118943Z   -s, --seeds strings                       A comma-separated list of seed
2024-12-11T08:26:05.0119543Z                                             nodes to connect to
2024-12-11T08:26:05.0120067Z                                             (scheme://host:port|name)
2024-12-11T08:26:05.0120691Z       --timeout duration                    The maximum time to wait for
2024-12-11T08:26:05.0121318Z                                             the checks and tune processes
2024-12-11T08:26:05.0121837Z                                             to complete (e.g. 300ms, 1.5s,
2024-12-11T08:26:05.0122344Z                                             2h45m) (default 10s)
2024-12-11T08:26:05.0123082Z       --tune                                When present will enable
2024-12-11T08:26:05.0123568Z                                             tuning before starting redpanda
2024-12-11T08:26:05.0124096Z       --well-known-io string                The cloud provider and VM
2024-12-11T08:26:05.0124677Z                                             type, in the format
2024-12-11T08:26:05.0125178Z                                             <provider>:<vm type>:<storage type>
2024-12-11T08:26:05.0125492Z 
2024-12-11T08:26:05.0125607Z Global Flags:
2024-12-11T08:26:05.0126101Z       --config string            Redpanda or rpk config file; default
2024-12-11T08:26:05.0126602Z                                  search paths are
2024-12-11T08:26:05.0127091Z                                  "/var/lib/redpanda/.config/rpk/rpk.yaml",
2024-12-11T08:26:05.0127612Z                                  $PWD/redpanda.yaml, and
2024-12-11T08:26:05.0128101Z                                  /etc/redpanda/redpanda.yaml
2024-12-11T08:26:05.0128693Z   -X, --config-opt stringArray   Override rpk configuration settings; '-X
2024-12-11T08:26:05.0129300Z                                  help' for detail or '-X list' for terser
2024-12-11T08:26:05.0129759Z                                  detail
2024-12-11T08:26:05.0130191Z       --profile string           rpk profile to use
2024-12-11T08:26:05.0130698Z   -v, --verbose                  Enable verbose logging
2024-12-11T08:26:05.0131049Z 
```

**Comment by kie-ci3** [comment]
**PR job** `#1` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-kogito-pipelines/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-kogito-runtimes -u https://github.com/apache/incubator-kie-kogito-runtimes/pull/3812 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3812/1/display/redirect

**Test results:**
- PASSED: 3364
- FAILED: 8

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3812/1/testReport/org.kie.kogito.persistence.kafka/KafkaProcessInstancesIT/testValuesReadMode/">org.kie.kogito.persistence.kafka.KafkaProcessInstancesIT.testValuesReadMode</a></summary>
Container startup failed for image redpandadata/redpanda:v24.3.1
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3812/1/testReport/org.kie.kogito.persistence.kafka/KafkaProcessInstancesIT/testFindByIdReadMode/">org.kie.kogito.persistence.kafka.KafkaProcessInstancesIT.testFindByIdReadMode</a></summary>
Container startup failed for image redpandadata/redpanda:v24.3.1
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3812/1/testReport/org.kie.kogito.quarkus.workflows/AsyncAPIIT/testConsumer/">org.kie.kogito.quarkus.workflows.AsyncAPIIT.testConsumer</a></summary>
java.util.concurrent.CompletionException: java.lang.RuntimeException: Unable to start Quarkus test resource class org.kie.kogito.testcontainers.quarkus.KafkaQuarkusTestResource
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3812/1/testReport/org.kie.kogito.quarkus.workflows/EmitEnumRPCGreetIT/testStateIsUnknown/">org.kie.kogito.quarkus.workflows.EmitEnumRPCGreetIT.testStateIsUnknown</a></summary>
java.util.concurrent.CompletionException: java.lang.RuntimeException: Unable to start Quarkus test resource class org.kie.kogito.testcontainers.quarkus.KafkaQuarkusTestResource
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3812/1/testReport/org.kie.kogito.it/KafkaPersistenceIT/testPersistence/">org.kie.kogito.it.KafkaPersistenceIT.testPersistence</a></summary>
java.util.concurrent.CompletionException: java.lang.RuntimeException: Unable to start Quarkus test resource class org.kie.kogito.testcontainers.quarkus.KafkaQuarkusTestResource
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3812/1/testReport/org.kie.kogito.integrationtests.quarkus/AdHocFragmentsIT/testUserTaskProcess/">org.kie.kogito.integrationtests.quarkus.AdHocFragmentsIT.testUserTaskProcess</a></summary>
java.util.concurrent.CompletionException: java.lang.RuntimeException: Unable to start Quarkus test resource class org.kie.kogito.testcontainers.quarkus.KafkaQuarkusTestResource
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3812/1/testReport/org.kie.kogito.integrationtests.springboot/PingPongMessageTest/verifyTopicsInformation/">org.kie.kogito.integrationtests.springboot.PingPongMessageTest.verifyTopicsInformation</a></summary>
Failed to load ApplicationContext for [WebMergedContextConfiguration@2ceb722e testClass = org.kie.kogito.integrationtests.springboot.PingPongMessageTest, locations = [], classes = [org.kie.kogito.integrationtests.springboot.KogitoSpringbootApplication], contextInitializerClasses = [org.kie.kogito.testcontainers.springboot.KafkaSpringBootTestResource], activeProfiles = [], propertySourceDescriptors = [], propertySourceProperties = ["org.springframework.boot.test.context.SpringBootTestContextBootstrapper=true", "server.port=0"], contextCustomizers = [org.springframework.boot.test.context.filter.ExcludeFilterContextCustomizer@2caf6912, org.springframework.boot.test.json.DuplicateJsonObjectContextCustomizerFactory$DuplicateJsonObjectContextCustomizer@aed0151, org.springframework.boot.test.mock.mockito.MockitoContextCustomizer@0, org.springframework.boot.test.web.client.TestRestTemplateContextCustomizer@4e558728, org.springframework.boot.test.autoconfigure.actuate.observability.ObservabilityContextCustomizerFactory$DisableObservabilityContextCustomizer@1f, org.springframework.boot.test.autoconfigure.properties.PropertyMappingContextCustomizer@0, org.springframework.boot.test.autoconfigure.web.servlet.WebDriverContextCustomizer@3605c4d3, org.springframework.boot.test.context.SpringBootTestAnnotation@2e20d0b5], resourceBasePath = "src/main/webapp", contextLoader = org.springframework.boot.test.context.SpringBootContextLoader, parent = null]
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3812/1/testReport/org.kie.kogito.integrationtests.springboot/PingPongMessageTest/testPingPongBetweenProcessInstances/">org.kie.kogito.integrationtests.springboot.PingPongMessageTest.testPingPongBetweenProcessInstances</a></summary>
ApplicationContext failure threshold (1) exceeded: skipping repeated attempt to load context for [WebMergedContextConfiguration@2ceb722e testClass = org.kie.kogito.integrationtests.springboot.PingPongMessageTest, locations = [], classes = [org.kie.kogito.integrationtests.springboot.KogitoSpringbootApplication], contextInitializerClasses = [org.kie.kogito.testcontainers.springboot.KafkaSpringBootTestResource], activeProfiles = [], propertySourceDescriptors = [], propertySourceProperties = ["org.springframework.boot.test.context.SpringBootTestContextBootstrapper=true", "server.port=0"], contextCustomizers = [org.springframework.boot.test.context.filter.ExcludeFilterContextCustomizer@2caf6912, org.springframework.boot.test.json.DuplicateJsonObjectContextCustomizerFactory$DuplicateJsonObjectContextCustomizer@aed0151, org.springframework.boot.test.mock.mockito.MockitoContextCustomizer@0, org.springframework.boot.test.web.client.TestRestTemplateContextCustomizer@4e558728, org.springframework.boot.test.autoconfigure.actuate.observability.ObservabilityContextCustomizerFactory$DisableObservabilityContextCustomizer@1f, org.springframework.boot.test.autoconfigure.properties.PropertyMappingContextCustomizer@0, org.springframework.boot.test.autoconfigure.web.servlet.WebDriverContextCustomizer@3605c4d3, org.springframework.boot.test.context.SpringBootTestAnnotation@2e20d0b5], resourceBasePath = "src/main/webapp", contextLoader = org.springframework.boot.test.context.SpringBootContextLoader, parent = null]
</details>

**Comment by gmunozfe** [comment]
@tkobayas take a look at this fix https://github.com/tkobayas/kogito-runtimes/pull/2

**Comment by tkobayas** [comment]
Thank you! @gmunozfe

**Comment by tkobayas** [comment]
Need fixes on other repos...

**Review by fjtirado — APPROVED** [review]

**Review by yesamer — APPROVED** [review]

**Pull request merged**